lib/PageInteractions.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: TBD
+# Maintainer: Oleksandr Orlov <oorlov@suse.de>
+
+package PageInteractions;
+use strict;
+use warnings FATAL => 'all';
+
+use testapi;
+use parent 'Exporter';
+use Exporter;
+
+our @EXPORT = qw(
+    find
+    send_keys
+    fill_in_password
+    is_displayed
+);
+
+sub find {
```

Comment:
It's more like `wait`. I'm not sure about this particular thing.

Reply:
My initial idea was to make something like find('needle_of_the_screen_where_action_should_be_performed')->send_key('smth'), mainly to sort of _force_ test writers to carefully check where the test is currently running before performing an action (that causes a lot of troubles currently).

My initial plan was to make testapi_extended class, that would be a child of testapi  with the 'find' method, that returns something like FoundArea class, where all the actions could be done. But current os-autoinst implementation don't allow to do this, as testapi is a perl package, not the class. Also, it is not really clear what to pass to FoundArea class as os-autoinst also not designed that way. So I just added this find() method as-is to show my vision on how the framework might be extended.

But for now, I agree that name for the method might be different (_ensure_visible_ for example). Again, I just want to ask test developers to check a screen before performing an action.